hs_core/search_indexes.py

```diff
@@ -764,303 +752,218 @@ def prepare_owners_count(self, obj):
-    #     """Return count of users who can view resource, otherwise return 0."""
-    #     if hasattr(obj, 'raccess'):
-    #         return obj.raccess.view_users.all().count()
-    #     else:
-    #         return 0
-
-    # def prepare_editor_login(self, obj):
-    #     """Return usernames of editors of a resource, otherwise return 0."""
-    #     if hasattr(obj, 'raccess'):
-    #         return [editor.username for editor in obj.raccess.edit_users.all()]
-    #     else:
-    #         return 0
-
-    # def prepare_editor(self, obj):
-    #     """Return full names of editors of a resource, otherwise return empty array."""
-    #     names = []
-    #     if hasattr(obj, 'raccess'):
-    #         for editor in obj.raccess.edit_users.all():
-    #             name = editor.last_name + ', ' + editor.first_name
-    #             names.append(name)
-    #     return names
-
-    # def prepare_editors_count(self, obj):
-    #     """Return count of editors of a resource, otherwise return 0."""
-    #     if hasattr(obj, 'raccess'):
-    #         return obj.raccess.edit_users.all().count()
-    #     else:
-    #         return 0
-
     # TODO: These should probably be multi-value fields and pick up all types.
     def prepare_geometry_type(self, obj):
         """
         Return geometry type if metadata exists, otherwise return [].
+        TODO: there can be multiples of these now.
         """
-        if hasattr(obj, 'metadata'):
-            if isinstance(obj.metadata, GeographicFeatureMetaData):
-                geometry_info = obj.metadata.geometryinformation
-                if geometry_info is not None:
-                    return geometry_info.geometryType
-                else:
-                    return None
-            else:
-                return None
-        else:
-            return None
+        for f in obj.geofeaturelogicalfile_set.all():
+            geometry_info = f.metadata.geometryinformation
+            if geometry_info is not None:
+                return geometry_info.geometryType
+        return None
 
     def prepare_field_name(self, obj):
         """
         Return metadata field name if exists, otherwise return [].
+        TODO: there can be multiples of these now.
         """
-        if hasattr(obj, 'metadata'):
-            if isinstance(obj.metadata, GeographicFeatureMetaData):
-                field_info = obj.metadata.fieldinformations.all().first()
-                if field_info is not None and field_info.fieldName is not None:
-                    return field_info.fieldName.strip()
-                else:
-                    return None
-            else:
-                return None
-        else:
-            return None
+        for f in obj.geofeaturelogicalfile_set.all():
+            field_info = f.metadata.fieldinformations.all().first()
+            if field_info is not None and field_info.fieldName is not None:
+                return field_info.fieldName.strip()
```

Comment:
Along the same lines, why only include the first of each logical file type?

Reply:
Same answer: I need to preserve the schema.